proto/pulumi/provider.proto

```diff
@@ -252,7 +260,12 @@ message DiffRequest {
     string id = 1;                     // the ID of the resource to diff.
     string urn = 2;                    // the Pulumi URN for this resource.
     google.protobuf.Struct olds = 3;   // the old output values of resource to diff.
-    google.protobuf.Struct news = 4;   // the new input values of resource to diff.
+
+    // the new input values of resource to diff.
+    //
+    // Pre-processed to accommodate ignoreChanges, see the note on CheckRequest.news.
+    google.protobuf.Struct news = 4;
```

Comment:
Is it the case that `DiffRequest.news` is pre-processed (separately) or is it the case that `news` will always be the output of `Check` and `Check` has `ignoreChanges` pre-applied?

In other words, can `Check` blindly *override* `ignoreChanges` on behalf of diff?

Reply:
Hmm I think you're right that this might just need a comment that these are checked inputs that Check returned. There's likely no additional processing. If Check does something differently it will be sent here as well. 

Check cannot truly opt out of ignoreChanges handling in the engine because the original pristine program level data is not made available to the provider, so if there's interesting values in the user program but the engine thinks they should be overwritten because of ignoreChanges, they will get overwritten. I'm double checking TF sets here, somewhere where we could possibly want to opt out of index-based ignoreChanges handling by the engine but cannot do so.

Reply:
These are checked inputs. No additional processing happens engine side as far as I recall.